docs/zkapps/advanced-snarkyjs/permissions.mdx

```diff
@@ -0,0 +1,290 @@
+    let senderUpdate = AccountUpdate.createSigned(this.sender);
+    senderUpdate.send({ to: this, amount });
+  }
+}
+```
+
+
+Our `UnsecureContract` has two methods: `withdraw` for withdrawing funds from the smart contract account and `deposit` for depositing funds into the smart contract account.
+Also notice how we specify our permissions in the `init` method and set the permission for `send` to `Permissions.none()`? 
+Because `none` means we don't have to provide *any* form of authorization, a malicious actor can easily drain all funds from the smart contract! Take a look at the following transaction that abuses that:
+
+
+```ts
+tx = await Mina.transaction(account1Address, () => {
+  let withdrawal = AccountUpdate.create(zkappAddress);
+  withdrawal.send({ to: account1Address, amount: 1e9 });
+});
+await tx.sign([account1Key]).send();
+```
+
+The transaction above creates a new account update for our smart contract address. Right after that, we tell our new account update to send 1 MINA to the address of our fee payer (account1Address).
+At the end of our transaction block, we only sign the transaction with the private key of the fee payer - not the private key of the smart contract!
+Because our permissions for sending funds away from a smart contract were set to `none`, this transaction will succeed and drain 1 MINA from the smart contract.
+
+If we now change our `send` permission to something else, like `signature`,
+our manual account update will fail with `Update_not_permitted_balance` which means we are not allowed to withdraw funds from the smart contract,
+since our authorization does not fit the permission for `send` - which now requires a valid signature!
+
+We can now slightly modify our withdraw-transaction to include a valid signature by adding requesting a signature on our `withdrawal` account update and providing the private key of the smart contract account to `tx.send([zkappKey]) `
+
+```ts
+tx = await Mina.transaction(account1Address, () => {
+  let withdrawal = AccountUpdate.create(zkappAddress);
+  withdrawal.send({ to: account1Address, amount: 1e9 });
+  withdrawal.requireSignature();
+});
+await tx.sign([account1Key, zkappKey]).send();
+```
+
+Now that we have provided a valid signature and the smart contract requires a valid signatures for sending funds from the smart contract, our transaction will succeed!
+
+
+### Upgradeability of smart contracts
+
+Another important part of smart contract development is upgradeability.
+By using permissions, we can either make our smart contract not upgradeable at all or upgradeable! 
+On Mina, when we deploy a smart contract we generate a verification key from our contract source code, which will then be stored on-chain and used to verify proofs that belong to that smart contract.
+We remember that one permission was called `setVerificationKey` - by modifying the authorization kind for this permission, we can make our smart contract upgradeable or not. 
```

Comment:
It's great that you added the upgradability section! I think, though, that the example should also set `setPermissions` to impossible. Only that will make the contract truly frozen